include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce_then_scan.h

```diff
@@ -659,15 +665,17 @@ struct __get_bounds_simple
         const auto& __rng1 = std::get<0>(__in_rng.tuple()); // first sequence
         const auto& __rng2 = std::get<1>(__in_rng.tuple()); // second sequence
 
-        using _SizeType = decltype(__rng1.size());
-        return std::make_tuple(_SizeType{0}, __rng1.size(), _SizeType{0}, __rng2.size());
+        using _SizeType = std::common_type_t<std::ranges::range_size_t<decltype(__rng1)>, std::ranges::range_size_t<decltype(__rng2)>>;
```

Comment:
```suggestion
        using _SizeType = std::common_type_t<std::make_unsigned_t<decltype(__rng1.size())>, std::make_unsigned_t<decltype(__rng2.size())>>;
```

We probably should make a `__dpl_ranges_size_t<R>`, but I'm not sure if we want to try to do that in this PR.  It probably deserves its own PR after the milestone.  For now, we need to do it like this to avoid c++17 issues.